### PR TITLE
BTM-237: Add stack-teardown script.

### PR DESCRIPTION
**Pull request by phinterseer**

### CODE REVIEW

**Comment by MelAtGDS** [comment]
Thanks Peter. Is it worth updating the README on this as well?

**Comment by phinterseer** [comment]
Ah, absolutely, thanks Mel! :-)

**Comment by NKannan14** [comment]
@phinterseer  just noticed the message for deleting the resource manually it says ignore s3 objects. So just manually deleting the Athena workgroup, removes the stack successfully  
image
image

**Review by MelAtGDS — APPROVED** [review]

File: src/tools/teardown-cf-stack.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+const stackName = `di-btm-${process.env.ENV_NAME}`;
+
+const wait = (milliseconds) => {
+  return new Promise(resolve => setTimeout(resolve, milliseconds));
+}
+
+const deleteObject = async (bucket, objectName, version) => {
+  const result = await s3Client.send(new DeleteObjectCommand({Bucket: bucket, Key: objectName, VersionId: version})).catch(err => {return { Error: err }});
+  // console.log('deleteObject: ', result);
```
[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.js:44]: That's for easy debugging when something goes wrong. I figured it's OK to leave it in as it's a script, but I can remove it if it bothers you?

File: src/tools/teardown-cf-stack.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+if (!stackSuffix) {
+  console.log('Please set the environment variable ENV_NAME to specify the stack to delete.');
+  process.exit(-1);
+}
+
+if (stackSuffix.slice(0,4) !== 'dev-') {
+  console.log('This script can only be used to delete dev-* stacks.');
+  process.exit(-1);
+}
```
[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.js:34]: Limiting it here for use in dev and the correct namespace names to limit accidental use in other environments and to enforce the naming convention, at least a little. You can always disable the checks if you need to use it for something else.

**Pull request merged**